asapdiscovery-ml/asapdiscovery/ml/utils.py

```diff
@@ -1220,6 +1244,171 @@ def split_molecules(ds, split_fracs, generator=None):
     return all_subsets
 
 
+def split_temporal(ds, split_fracs, grouped=False, reverse=False, end_splits=2):
+    """
+    Split molecules temporally by date created. Earlier molecules will be placed in the
+    training set and later molecules will be placed in the val/test sets (unless
+    `reverse` is set).
+
+    Parameters
+    ----------
+    ds : Union[cml.data.DockedDataset, cml.data.GraphDataset]
+        Molecular dataset to split
+    split_fracs : List[float]
+        List of fraction of compounds to put in each split
+    grouped : bool, default=False
+        Splitting a GroupedDockedDataset object
+    reverse : bool, default=False
+        Reverse sorting of data
+    end_splits : int, default=2
```

Comment:
I'm not exactly sure how this works and what a default of `2` corresponds to. I think a practical example will help.

Reply:
the idea for a default of 2 is that we'd want to take the val and test splits from the end, but I've changed the default to 0 and passed the responsibility of setting this arg into the `split_dataset` dispatch function